WebDriverAgentLib/Categories/XCElementSnapshot+FBHelpers.m

```diff
@@ -57,9 +57,21 @@ - (id)fb_attributeValue:(NSNumber *)attribute
   return (id __nonnull)attributesResult[attribute];
 }
 
+inline static BOOL valuesAreEqual(id value1, id value2);
+
+inline static BOOL isNilOrEmpty(id value);
+
 - (BOOL)fb_framelessFuzzyMatchesElement:(XCElementSnapshot *)snapshot
 {
-  return [self.wdUID isEqualToString:snapshot.wdUID];
+  if (isNilOrEmpty(self.identifier) && isNilOrEmpty(self.title) && isNilOrEmpty(self.label) &&
```

Comment:
Please add an explanation comment on why it is needed to compare snapshots this way